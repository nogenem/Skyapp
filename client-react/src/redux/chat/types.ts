import { TMessageType } from '~/constants/message_types';
import { TUserStatus } from '~/constants/user_status';

interface IOtherUser {
  _id: string;
  nickname: string;
  thoughts: string;
  status: TUserStatus;
  online: boolean;
  channel_id?: string;
}

interface IOtherUsers {
  [_id: string]: IOtherUser;
}

interface IMember {
  user_id: string;
  is_adm: boolean;
  last_seen: Date;
}

interface IChannel {
  _id: string;
  name: string;
  is_group: boolean;
  members: IMember[];
  other_member_idx?: number;
  unread_msgs: number;
  lastMessage?: IMessage;
}

interface IChannels {
  [_id: string]: IChannel;
}

interface IAttachment {
  originalName: string;
  size: number;
  path: string;
  mimeType: string;
  imageDimensions?: { width: number; height: number };
}

interface IMessage {
  _id: string;
  channel_id: string;
  from_id?: string;
  body: string | IAttachment;
  type: TMessageType;
  createdAt: Date;
  updatedAt: Date;
}

interface IInitialData {
  users: IOtherUsers;
  channels: IChannels;
}

interface IActiveChannelInfo {
  _id: string;
  messages: IMessage[];
  totalMessages: number;
  queue: IMessage[];
}

interface INewGroupCredentials {
  name: string;
  members: string[];
  admins: string[];
}

interface IUpdateGroupCredentials {
  channel_id: string;
  name: string;
  members: string[];
  admins: string[];
}

interface ILeaveGroupCredentials {
  channel_id: string;
}

interface IFetchMessagesCredentials {
  channel_id: string;
  offset: number;
  limit?: number;
  sort?: string;
}

interface ISendMessageCredentials {
  channel_id: string;
  body: string;
}

type TChatState = {
  users: IOtherUsers;
  channels: IChannels;
  activeChannelInfo?: IActiveChannelInfo;
};

enum EChatActions {
  SET_INITIAL_DATA = '@chat/SET_INITIAL_DATA',
  SET_USER_ONLINE = '@chat/SET_USER_ONLINE',
  ADD_OR_UPDATE_CHANNEL = '@chat/ADD_OR_UPDATE_CHANNEL',
  SET_ACTIVE_CHANNEL = '@chat/SET_ACTIVE_CHANNEL',
  REMOVE_CHANNEL = '@chat/REMOVE_CHANNEL',
  ADD_NEW_USER = '@chat/ADD_NEW_USER',
  ADD_MESSAGES = '@chat/ADD_MESSAGES',
  SET_LATEST_MESSAGE = '@chat/SET_LATEST_MESSAGE',
  ADD_TO_MESSAGES_QUEUE = '@chat/ADD_TO_MESSAGES_QUEUE',
  REMOVE_FROM_MESSAGES_QUEUE = '@chat/REMOVE_FROM_MESSAGES_QUEUE',
  SET_LAST_SEEN = '@chat/SET_LAST_SEEN',
  SET_USER_STATUS = '@chat/SET_USER_STATUS',
  SET_USER_THOUGHTS = '@chat/SET_USER_THOUGHTS',
}

interface IChatActionType<T, P> {
  type: T;
  payload: P;
}

type TChatAction =
  | IChatActionType<typeof EChatActions.SET_INITIAL_DATA, IInitialData>
  | IChatActionType<
      typeof EChatActions.SET_USER_ONLINE,
      { _id: string; value: boolean }
    >
  | IChatActionType<typeof EChatActions.ADD_OR_UPDATE_CHANNEL, IChannel>
  | IChatActionType<
      typeof EChatActions.SET_ACTIVE_CHANNEL,
      { _id: string | undefined }
    >
  | IChatActionType<typeof EChatActions.REMOVE_CHANNEL, { channelId: string }>
  | IChatActionType<typeof EChatActions.ADD_NEW_USER, IOtherUser>
  | IChatActionType<
      typeof EChatActions.ADD_MESSAGES,
      { messages: IMessage[]; totalMessages: number; atTop: boolean }
    >
  | IChatActionType<typeof EChatActions.SET_LATEST_MESSAGE, IMessage>
  | IChatActionType<typeof EChatActions.ADD_TO_MESSAGES_QUEUE, IMessage>
  | IChatActionType<typeof EChatActions.REMOVE_FROM_MESSAGES_QUEUE, IMessage>
  | IChatActionType<
      typeof EChatActions.SET_LAST_SEEN,
      {
        channel_id: string;
        user_id: string;
        last_seen: string;
      }
    >
  | IChatActionType<
      typeof EChatActions.SET_USER_STATUS,
      {
        user_id: string;
        newStatus: TUserStatus;
      }
    >
  | IChatActionType<
      typeof EChatActions.SET_USER_THOUGHTS,
      {
        user_id: string;
        newThoughts: string;
      }
    >;

export type {
  TChatState,
  TChatAction,
  IInitialData,
  IOtherUsers,
  IOtherUser,
  IChannel,
  IChannels,
  IMember,
  IMessage,
  IAttachment,
  INewGroupCredentials,
  IUpdateGroupCredentials,
  ILeaveGroupCredentials,
  IFetchMessagesCredentials,
  ISendMessageCredentials,
};
export { EChatActions };
