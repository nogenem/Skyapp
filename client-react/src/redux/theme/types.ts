type TThemeMode = 'dark' | 'light';

type TThemeState = {
  mode?: TThemeMode;
};

enum EThemeActions {
  MODE_SWITCHED = '@theme/MODE_SWITCHED',
}

interface IThemeActionType<T, P> {
  type: T;
  payload: P;
}

type TThemeAction = IThemeActionType<
  typeof EThemeActions.MODE_SWITCHED,
  TThemeMode
>;

export type { TThemeMode, TThemeState, TThemeAction };
export { EThemeActions };
