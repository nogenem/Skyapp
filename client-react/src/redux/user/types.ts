interface User {
  _id: string;
  token?: string;
}

type UserState = User;

enum EUserActions {
  SIGNED_IN = '@user/SIGNED_IN',
  SIGNED_OUT = '@user/SIGNED_OUT',
}

interface UserActionType<T, P> {
  type: T;
  payload: P;
}

type UserAction =
  | UserActionType<typeof EUserActions.SIGNED_IN, User>
  | UserActionType<typeof EUserActions.SIGNED_OUT, null>;

export type { User, UserState, UserAction };
export { EUserActions };
