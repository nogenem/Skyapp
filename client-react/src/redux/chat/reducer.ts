import produce from 'immer';
import { createSelector } from 'reselect';

import type { IAppState } from '../store';
import { EChatActions } from './types';
import type { TChatAction, TChatState } from './types';

export const initialState: TChatState = {
  users: {},
  channels: {},
};

export default function chat(
  state = initialState,
  action: TChatAction,
): TChatState {
  return produce(state, draft => {
    switch (action.type) {
      case EChatActions.SET_INITIAL_DATA:
        draft.users = action.payload.users;
        draft.channels = action.payload.channels;
        return draft;
      case EChatActions.SET_USER_ONLINE:
        if (draft.users[action.payload._id]) {
          draft.users[action.payload._id].online = action.payload.value;
        }
        return draft;
      default:
        return draft;
    }
  });
}

// SELECTORS
export const getChat = (state: IAppState) => state.chat || initialState;
export const getUsers = createSelector(getChat, data => data.users);
export const getChannels = createSelector(getChat, data => data.channels);
