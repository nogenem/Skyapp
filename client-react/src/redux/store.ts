import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import rootReducer from './rootReducer';
import type { TThemeState } from './theme/types';
import type { TUserState } from './user/types';

interface IAppState {
  user: TUserState;
  theme: TThemeState;
}

let enhancer = applyMiddleware(thunk);
if (process.env.NODE_ENV !== 'production')
  enhancer = composeWithDevTools(enhancer);

export type { IAppState };
export default createStore(rootReducer, {}, enhancer);
