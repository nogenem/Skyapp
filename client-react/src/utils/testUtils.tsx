import React from 'react';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { IAppState } from '~/redux/store';

export const getMockStore = () => {
  const middlewares = [thunk];
  return configureStore(middlewares);
};

export const getRenderWithRedux = () => {
  const emptyState: Partial<IAppState> = {};
  const mockStore = getMockStore();

  return (
    ui: React.ReactNode,
    initialState: Partial<IAppState> = emptyState,
  ) => {
    const store = mockStore(initialState);
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
    };
  };
};
