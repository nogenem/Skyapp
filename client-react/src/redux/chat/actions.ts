import { Dispatch } from 'redux';

import * as SOCKET_EVENTS from '~/constants/socket_events';
import type { IUser } from '~/redux/user/types';
import api from '~/services/api';
import io from '~/services/io';

import type {
  IChannel,
  IInitialData,
  INewGroupCredentials,
  IOtherUser,
  IUpdateGroupCredentials,
} from './types';
import { EChatActions } from './types';

const setInitialData = (data: IInitialData) => ({
  type: EChatActions.SET_INITIAL_DATA,
  payload: data,
});

const setUserOnline = (_id: string, value: boolean) => ({
  type: EChatActions.SET_USER_ONLINE,
  payload: {
    _id,
    value,
  },
});

const addNewChannel = (channel: IChannel) => ({
  type: EChatActions.ADD_NEW_CHANNEL,
  payload: channel,
});

export const setActiveChannel = (channel_id: string) => ({
  type: EChatActions.SET_ACTIVE_CHANNEL,
  payload: { _id: channel_id },
});

export const connectIo = (user: IUser) => (dispatch: Dispatch) => {
  const instance = io.instance();
  instance.connect({ _id: user._id });

  instance.socket!.on(SOCKET_EVENTS.SOCKET_CONNECT, () => {
    instance.socket!.emit(
      SOCKET_EVENTS.IO_GET_INITIAL_DATA,
      (data: IInitialData) => {
        dispatch(setInitialData(data));
      },
    );

    instance.socket!.on(SOCKET_EVENTS.IO_SIGNIN, (_id: string) => {
      dispatch(setUserOnline(_id, true));
    });
    instance.socket!.on(SOCKET_EVENTS.IO_SIGNOUT, (_id: string) => {
      dispatch(setUserOnline(_id, false));
    });
    instance.socket!.on(
      SOCKET_EVENTS.IO_PRIVATE_CHANNEL_CREATED,
      (channel: IChannel) => {
        dispatch(addNewChannel(channel));
      },
    );
    instance.socket!.on(
      SOCKET_EVENTS.IO_GROUP_CHANNEL_CREATED,
      (channel: IChannel) => {
        dispatch(addNewChannel(channel));
      },
    );
  });
};

export const disconnectIo = () => (dispatch: Dispatch) => {
  io.instance().disconnect();
};

export const startChattingWith = (otherUser: IOtherUser) => (
  dispatch: Dispatch,
) =>
  api.chat.createChannelWith(otherUser).then(({ channel_id }) => {
    dispatch(setActiveChannel(channel_id));
  });

export const createGroupChannel = (credentials: INewGroupCredentials) => (
  dispatch: Dispatch,
) =>
  api.chat.createGroupChannel(credentials).then(({ channel_id }) => {
    dispatch(setActiveChannel(channel_id));
  });

export const updateGroupChannel = (credentials: IUpdateGroupCredentials) => (
  dispatch: Dispatch,
) =>
  api.chat.updateGroupChannel(credentials).then(({ channel_id }) => {
    dispatch(setActiveChannel(channel_id));
  });
