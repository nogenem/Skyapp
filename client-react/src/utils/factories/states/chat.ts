import faker from 'faker/locale/en_US';

import { initialState } from '~/redux/chat/reducer';
import type {
  IChannel,
  IChannels,
  IOtherUser,
  IOtherUsers,
  TChatState,
} from '~/redux/chat/types';

import activeChannelInfoFactory from '../models/activeChannelInfo';
import channelFactory from '../models/channel';
import otherUserFactory from '../models/otherUser';

interface IOptions {
  useConstValues: boolean;
  useInitialState: boolean;
  usersLen: number;
  channelsLen: number;
}

export default (
  override?: Partial<TChatState>,
  options?: Partial<IOptions>,
): TChatState => {
  let usersLen = faker.datatype.number({ min: 0, max: 2 });
  if (options?.usersLen !== undefined && options.usersLen >= 0) {
    usersLen = options.usersLen;
  } else if (options?.useConstValues) {
    usersLen = 2;
  }

  let tmpOtherUsers: IOtherUser[] = [];
  if (!options?.useConstValues) {
    tmpOtherUsers = Array.from({ length: usersLen }, () => otherUserFactory());
  } else {
    tmpOtherUsers = Array.from({ length: usersLen }, (_, idx) =>
      otherUserFactory(
        { _id: `other-user-${idx}`, channel_id: `channel-${idx}` },
        { useConstValues: true },
      ),
    );
  }

  const users: IOtherUsers = {};
  tmpOtherUsers.forEach(user => {
    users[user._id] = user;
  });

  let channelsLen = faker.datatype.number({ min: 0, max: 2 });
  if (options?.channelsLen !== undefined && options.channelsLen >= 0) {
    channelsLen = options.channelsLen;
  } else if (options?.useConstValues) {
    channelsLen = 2;
  }

  let tmpChannels: IChannel[] = [];
  if (!options?.useConstValues) {
    tmpChannels = Array.from({ length: channelsLen }, () => channelFactory());
  } else {
    tmpChannels = Array.from({ length: channelsLen }, (_, idx) =>
      channelFactory({ _id: `channel-${idx}` }, { useConstValues: true }),
    );
  }

  const channels: IChannels = {};
  tmpChannels.forEach(channel => {
    channels[channel._id] = channel;
  });

  let activeChannelInfo =
    Math.random() < 0.5 ? undefined : activeChannelInfoFactory();
  if (options?.useConstValues) activeChannelInfo = undefined;

  return options?.useInitialState
    ? initialState
    : ({
        users,
        channels,
        activeChannelInfo,
        ...(override || {}),
      } as TChatState);
};

export type { IOptions };
