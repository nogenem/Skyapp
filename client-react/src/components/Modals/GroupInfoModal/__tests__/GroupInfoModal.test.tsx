import React from 'react';

import { IOtherUser } from '~/redux/chat/types';
import { IAppState } from '~/redux/store';
import FACTORIES from '~/utils/factories';
import { getRenderWithRedux } from '~/utils/testUtils';

import { GroupInfoModal as ConnectedGroupInfoModal } from '../index';

const renderWithRedux = getRenderWithRedux();

describe('Connected GroupInfoModal', () => {
  it('renders correctly', () => {
    const channel = FACTORIES.models.channel({}, { useConstValues: true });
    const otherUser: IOtherUser = FACTORIES.models.otherUser(
      { channel_id: channel._id },
      { useConstValues: true },
    );
    const initialState: Partial<IAppState> = {
      chat: FACTORIES.states.chat({
        channels: { [channel._id]: channel },
        users: { [otherUser._id]: otherUser },
      }),
    };

    const { getByRole } = renderWithRedux(
      <ConnectedGroupInfoModal isOpen channel={channel} onClose={() => {}} />,
      initialState,
    );

    expect(getByRole('presentation')).toMatchSnapshot();
  });
});
