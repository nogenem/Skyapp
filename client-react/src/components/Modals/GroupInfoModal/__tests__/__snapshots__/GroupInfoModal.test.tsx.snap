// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected GroupInfoModal renders correctly 1`] = `
<div
  class="MuiDialog-root"
  role="presentation"
  style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper"
    role="none presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="group-info-modal-title"
      class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
      role="dialog"
    >
      <div
        class="MuiDialogTitle-root"
        id="group-info-modal-title"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6"
        >
          Messages:Group Info
        </h2>
      </div>
      <div
        class="MuiDialogContent-root MuiDialogContent-dividers"
      >
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-textInput-10 MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled Mui-required Mui-required"
              data-shrink="true"
              for="group-info-modal-form-name"
              id="group-info-modal-form-name-label"
            >
              Common:Group Name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                disabled=""
                id="group-info-modal-form-name"
                name="name"
                required=""
                type="text"
                value="Channel 1"
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-13 PrivateNotchedOutline-legendNotched-14"
                >
                  <span>
                    Common:Group Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <fieldset
            class="MuiFormControl-root makeStyles-formControl-2"
            id="members_wrapper"
          >
            <legend
              class="MuiFormLabel-root makeStyles-formLegend-4"
            >
              Messages:Select Members
            </legend>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-textInput-10 MuiFormControl-marginNormal MuiFormControl-fullWidth"
              style="margin-top: 5px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled"
                data-shrink="false"
                for="group-info-modal-form-filter"
                id="group-info-modal-form-filter-label"
              >
                Common:Filter
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled"
                  disabled=""
                  id="group-info-modal-form-filter"
                  name="filter"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <p
              class="MuiFormHelperText-root"
            >
              Messages:Click on user's image to toggle admin status
            </p>
            <div
              class="MuiFormGroup-root makeStyles-formGroup-3"
            >
              <label
                class="MuiFormControlLabel-root makeStyles-checkboxWrapper-6 MuiFormControlLabel-labelPlacementStart Mui-disabled"
              >
                <span
                  aria-disabled="true"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-disabled-17 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
                  tabindex="-1"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-18"
                      data-indeterminate="false"
                      disabled=""
                      name="1"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
                >
                  <span
                    class="makeStyles-checkboxLabel-7"
                  >
                    <span
                      class=""
                      title="Messages:Turn into Admin"
                    >
                      <button
                        aria-label="Messages:Turn into Admin"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                        data-testid="user-avatar"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <div
                            class="makeStyles-container-19"
                          >
                            <div
                              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root makeStyles-icon-20"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                                />
                              </svg>
                            </div>
                            <span
                              class="makeStyles-dotContainer-21"
                            >
                              <span
                                class="makeStyles-dot-22 makeStyles-dot_active-23 "
                                title="Common:Active"
                              />
                            </span>
                          </div>
                        </span>
                      </button>
                    </span>
                    <span>
                      Test
                    </span>
                  </span>
                </span>
              </label>
            </div>
            <p
              class="MuiFormHelperText-root makeStyles-formHelperText-5"
            />
          </fieldset>
        </form>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Common:Close
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
